src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+      if (idx < 0 || cigar.length <= idx) return stretchLength
+      cigar(idx) match {
+        case Match | CallEnd | BothEnd | TruthEnd => stretchLength
+        case Mismatch     => checkNext(cigar=cigar, idx=idx+by, by=by, stretchLength=stretchLength+1)
+        case Call | Truth => checkNext(cigar=cigar, idx=idx+by, by=by, stretchLength=stretchLength)
+      }
+    }
+
+    // Gets all the indices in the cigar where there is a long switch error. Currently returns the index for all cigar
+    // mismatches that contribute to a long switch error, and tries to maximize such a stretch.
+    val longSwitchErrorIndices = cigar.indices
+      .filter { i => cigar(i) == Mismatch }
+      .filter { i => 1 + checkNext(cigar, i-1, -1) + checkNext(cigar, i+1, 1) > 1 }
+
+    val numLongSwitchErrors = longSwitchErrorIndices.foldLeft((-2, 0)) { case ((lastIdx, count), idx) =>
+      if (lastIdx + 1 == idx) (idx, count)
+      else (idx, count+1)
+    }._2
+
+    val numLongSwitchSites = cigar.count {
+      case Mismatch | Match => true
+      case _                => false
+    }
+
+    (numLongSwitchErrors, numLongSwitchSites)
+  }
+
+  case class IlluminaSwitchErrors(var numPointErrors: Int, var numLongSwitchErrors: Int, var numSites: Int) {
+    require(numPointErrors <= numSites)
+    require(numLongSwitchErrors <= numSites)
+    def add(other: IlluminaSwitchErrors): this.type = {
+      this.numPointErrors += other.numPointErrors
+      this.numLongSwitchErrors += other.numLongSwitchErrors
+      this.numSites += other.numSites
+      this
+    }
+  }
+
+  /** Computes the number of point and long switch error rates as described in http://dx.doi.org/10.1038%2Fng.3119 */
+  def toIlluminaSwitchErrors(cigar: Cigar, pointPenalty: Int = -1, transitionPenalty: Int = -5): IlluminaSwitchErrors = {
+
+    // 1. partition the cigar into call blocks.
+    val subCigars = toPhasedBlocks(cigar=cigar, isTruth=false)
+
+    // 2. for each call block, run the HMM, and sum the # of long swith errors
+    subCigars.map { subCigar =>
+      toIlluminaSwitchErrorsHmm(cigar=subCigar, pointPenalty=pointPenalty, transitionPenalty=transitionPenalty)
+    }.foldLeft(IlluminaSwitchErrors(0, 0, 0)) { case (acc, cur) => acc.add(cur) }
+  }
+
+  /** Computes the number of point and long switch error rates as described in http://dx.doi.org/10.1038%2Fng.3119
+    *
+    * The HMM with a simple scoring system and two hidden states, one for each parental haplotype.  The emission
+    * probability is scored as follows: 0 for being on the correct haplotype, otherwise `pointPenalty`.  The transition
+    * probability is scored as follows: 0 for staying on the same haplotype, otherwise `transitionPenalty`.
+    *
+    * We prefer a long switch error over a sequence of point errors (i.e. when both produce have the same score). */
+  private[vcf] def toIlluminaSwitchErrorsHmm(cigar: Cigar, pointPenalty: Int = -1, transitionPenalty: Int = -5): IlluminaSwitchErrors = {
```

Comment:
@nh13 I feel like to properly review this method i need to read and fully understand the method from the paper, and that's going to take a while.  If you think that's necessary/worth the investment let me know and I'll do it, otherwise I'm skipping this method for now.